internal/host/transition_test.go

```diff
@@ -1507,6 +1509,176 @@ var _ = Describe("Refresh Host", func() {
+			hostID    strfmt.UUID
+			inventory string
+			role      models.HostRole
+			dstState  string
+			srcState  string
+		}{
+			{
+				name:          "insufficient worker memory",
+				hostID:        strfmt.UUID("054e0100-f50e-4be7-874d-73861179e40d"),
+				inventory:     hostutil.GenerateInventoryWithResourcesWithBytes(4, 104857600, "worker"),
+				role:          models.HostRoleWorker,
+				srcState:      models.HostStatusDiscovering,
+				dstState:      models.HostStatusInsufficient,
+				statusInfoMsg: "Insufficient memory to deploy CNV. Required memory is: 360 MiB but found: 100 MiB",
+				validationsChecker: makeJsonChecker(map[validationID]validationCheckResult{
+					IsConnected:                 {status: ValidationSuccess, messagePattern: "Host is connected"},
+					IsHostnameValid:             {status: ValidationSuccess, messagePattern: ""},
+					IsAPIVipConnected:           {status: ValidationSuccess, messagePattern: "API VIP connectivity success"},
+					BelongsToMajorityGroup:      {status: ValidationPending, messagePattern: "Machine Network CIDR or Connectivity Majority Groups missing"},
+					HasInventory:                {status: ValidationSuccess, messagePattern: "Valid inventory exists for the host"},
+					HasMinCPUCores:              {status: ValidationSuccess, messagePattern: "Sufficient CPU cores"},
+					HasMinMemory:                {status: ValidationFailure, messagePattern: "The host is not eligible to participate in Openshift Cluster bec"},
+					HasMinValidDisks:            {status: ValidationSuccess, messagePattern: "Sufficient disk capacity"},
+					IsMachineCidrDefined:        {status: ValidationSuccess, messagePattern: "Machine Network CIDR is defined"},
+					HasCPUCoresForRole:          {status: ValidationSuccess, messagePattern: "Sufficient CPU cores for role worker"},
+					HasMemoryForRole:            {status: ValidationFailure, messagePattern: "Require at least 8 GiB RAM role worker, found only 0"},
+					IsHostnameUnique:            {status: ValidationSuccess, messagePattern: "Hostname worker is unique in cluster"},
+					BelongsToMachineCidr:        {status: ValidationSuccess, messagePattern: "Host belongs to machine network CIDR 1.2.3.0/24"},
+					IsPlatformValid:             {status: ValidationSuccess, messagePattern: "Platform RHEL is allowed"},
+					IsNTPSynced:                 {status: ValidationFailure, messagePattern: "Host couldn't synchronize with any NTP server"},
+					AreContainerImagesAvailable: {status: ValidationPending, messagePattern: "Missing container images statuses"},
+					AreLsoRequirementsSatisfied: {status: ValidationSuccess, messagePattern: ""},
+					AreOcsRequirementsSatisfied: {status: ValidationSuccess, messagePattern: "ocs is disabled"},
+					AreCnvRequirementsSatisfied: {status: ValidationFailure, messagePattern: "Insufficient memory to deploy CNV. Required memory is: 360 MiB but found: 100 MiB"},
+				}),
+			},
+			{
+				name:          "insufficient master memory",
```

Comment:
I think you can memory and CPU cases together since the validation is done in parallel, meaning that you should get the same outcomes whether the host has sufficient cpu but not memory and vice versa. WDYT?

Reply:
It is valid test case. Do you want me to update this PR or could I do it in the next one?

Reply:
I'm fine as it is right now. It can always be revisited later on.